src/boot/efi/recovery.c

```diff
@@ -0,0 +1,23 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "recovery.h"
+#include "util.h"
+
+#define FALLBACK_BOOT_ENTRY L"ostree-1-lmp.conf\0"
+
+bool recovery_check_firmware_failure(void) {
+        EFI_SYSTEM_RESOURCE_TABLE *esrt_table = NULL;
+        EFI_SYSTEM_RESOURCE_ENTRY *esrt_entry = NULL;
+
+        esrt_table = find_configuration_table(MAKE_GUID_PTR(EFI_SYSTEM_RESOURCE_TABLE));
+        if (esrt_table) {
+                esrt_entry = (void *)(esrt_table + 1);
```

Comment:
if you do +1 here, this always suggests to the reader that we are talking of an array here, but we aren't really, you just skip the header. hence i'd prefer if you'd write this like this:

```c
esrt_entry = (const EFI_SYSTEM_RESOURCE_ENTRY*) ((const uint8_t*) esrt_table + sizeof(EFI_SYSTEM_RESOURCE_TABLE));
```

even if this is longer it's immediately clear what this is